### PR TITLE
Add Dockerfile And K8s YAML

### PR DESCRIPTION
**Pull request by ihcsim**
This PR adds a Dockerfile to build the controller's image. It also introduces a set of YAML manifests which define the K8s resources needed to run the controller on Kubernetes. The new README has instructions on how to build the controller's binary and image.

All the EKS scripts are moved to a new subfolder in `deploy/eks`.

### CODE REVIEW

File: deploy/controller/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  template:
+    metadata:
+      labels:
+        app: diffsnap
+        role: controller
+    spec:
+      serviceAccountName: diffsnap-controller
+      containers:
+      - name: controller
+        image: quay.io/isim/diffsnap-controller:latest
```
Comment (ihcsim) [inline deploy/controller/deploy.yaml:29]: Replace this with your own image.

File: deploy/controller/rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: diffsnap-controller
+  namespace: diffsnap
+rules:
+- apiGroups: ["differentialsnapshot.k8s.io"]
```
Comment (ihcsim) [inline deploy/controller/rbac.yaml]: The group suffix might need to be renamed to an unprotected one.

Comment (phuongatemc) [inline deploy/controller/rbac.yaml]: We are switchining to use "sample.com" instead of "k8s.io" for the prototype.

**Review by phuongatemc — APPROVED** [review]
Approve to merge.  We can fix the suffix later.

**Pull request merged**